pkg/plugins/policies/meshhttproute/xds/configurer.go

```diff
@@ -22,22 +24,34 @@ func (c RoutesConfigurer) Configure(virtualHost *envoy_route.VirtualHost) error
 	matches := c.routeMatch(c.Matches)
 
 	for _, match := range matches {
-		r := &envoy_route.Route{
-			Match: match.routeMatch,
-			Action: &envoy_route.Route_Route{
+		rb := &route.RouteBuilder{}
+
+		rb.Configure(route.RouteMustConfigureFunc(func(envoyRoute *envoy_route.Route) {
+			// todo: create configurers for Match and Action
```

Comment:
I'm not quite sure the configurer pattern is necessary, it's very noisy IMO, does it bring any advantages? Why aren't simple functions `configure(KumaAPI, *EnvoyAPI)` enough?

Reply:
I don't mean to block the PR with this one, I suppose I was just curious for other opinions.

Reply:
TBH I like this pattern, I find it easier to read, for example instead of nesting functions:
```go
func configure(...) {
    configureRoute(...)
}

func configureRoute(...) {
   configureAction(...)
   configureRedirect(...)
}

func configureAction(...) {
   configureClusterSpecifier(...)
}
```
you can have
```go
NewRouteBuilder().
    Configure(route.NewAction().
        Configure(NewClusterSpecifier(...))).
    Configure(route.NewRedirect(...)
```
so you don't have to jump deep inside functions to understand the overall configuration. 

Also, it's easier to reuse things, especially given the fact that MeshGatewayRoute and MeshHTTPRoute have a lot in common.

Reply:
We'll discuss this pattern later, resolving for now